File: Advanced Methods of Programming/Toy Language Interpreter/src/Model/Expressions/Constant.java
package Model.Expressions;

import Model.ProgramState;

public class Constant implements IExpression {

    private Integer value;

    public Constant(int value){
        this.value = value;
    }

    @Override
    public int eval(ProgramState state) {
        return value;
    }

    public String toString(){
        return value.toString();
    }
}

File: Advanced Methods of Programming/CSharp Toy Language Interpreter/ToyLanguageInterpreter/Src/Model/Expressions/Constant.cs
using System;
using ToyLanguageInterpreter.Src.Model;
using ToyLanguageInterpreter.Src.Model.Expressions;

namespace ToyLanguageInterpreter.Src.View
{
    internal class Constant : IExpression
    {
        private int value;

        public Constant(int value)
        {
            this.value = value;
        }

        public int Eval(ProgramState state)
        {
            return value;
        }

        public override string ToString()
        {
            return value.ToString();
        }
    }
}